### ISSUE TITLE
Question: which android version is used?

### ISSUE THREAD

**Issue by PerfectCarl** [dicej/android-libcore64#3]

**Comment by dicej** [comment]
Git commit 995537af19719bf8c62f19b6df28d35ab52a5c4b.  It (probably) doesn't correspond to any official release -- that's just where master was at when I cloned https://android.googlesource.com/platform/libcore.  We can try rebasing against something newer and/or official.

**Comment by PerfectCarl** [comment]
https://android.googlesource.com/platform/libcore/+/995537af19719bf8c62f19b6df28d35ab52a5c4b
I am having a hard time to find out the android version that corresponds to that commit.

According to the commit date and the tag date, I would say android-4.2.2_r1.2.
What do you think?

**Comment by dicej** [comment]
On Fri, 25 Apr 2014, PerfectCarl wrote:

> https://android.googlesource.com/platform/libcore/+/995537af19719bf8c62f19b6df28d35ab52a5c4b
> I am having a hard time to find out the android version that corresponds to that commit.
> 
> According to the commit date and the tag date, I would say android-4.2.2_r1.2.
> What do you think?

  $ git clone https://android.googlesource.com/platform/libcore
  $ git tag --contains 995537af19719
android-4.4.1_r1
android-4.4.2_r1
android-4.4.2_r2
android-4.4_r0.7
android-4.4_r0.8
android-4.4_r0.9
android-4.4_r1
android-4.4_r1.1
android-4.4_r1.2
android-cts-4.4_r1
android-sdk-4.4.2_r1

That suggests that it's older than android-4.4_r0.7, but newer than any 
other tag (i.e. after 4.3).

**Comment by PerfectCarl** [comment]
Thanks for the answer (and thanks for the complete git command)

**Comment by PerfectCarl** [comment]
Just how out of sheer curiosity how log does a full rebase against a new android version take ?
Is it only about fixing obvious conflicts spotted by git tools, or is there any real risks to introduce major instabilities?

**Comment by dicej** [comment]
On Fri, 2 May 2014, PerfectCarl wrote:

> Just how out of sheer curiosity how log does a full rebase against a new 
> android version take ? Is it only about fixing obvious conflicts spotted 
> by git tools, or is there any real risks to introduce major 
> instabilities?

The rebases I've done so far have been pretty easy, but that was before 
all the Windows changes were added.  Also, it's been a long time since we 
last rebased, so it could get pretty hairy depending on how much has 
changed since then.  I'd guess that any problems not found by Git would be 
found by running the libcore test suite and the Avian test suite.  If 
everything passes, we can be pretty confident that we avoided regressions.
